bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/camera/CameraHandler.java

```diff
@@ -69,93 +102,136 @@ protected State getNAThingProperty(@NonNull String channelId) {
-        return resultStringBuilder.toString();
+        return Optional.of(resultStringBuilder.toString());
     }
 
-    @SuppressWarnings("null")
-    protected String getVpnUrl() {
-        return (module == null) ? null : module.getVpnUrl();
+    private Optional<String> getVpnUrl() {
+        return getModule().map(NAWelcomeCamera::getVpnUrl);
     }
 
-    public String getStreamURL(String videoId) {
-        String result = getVpnUrl();
-        if (result == null) {
-            return null;
+    public Optional<String> getStreamURL(String videoId) {
+        Optional<String> result = getVpnUrl();
+        if (!result.isPresent()) {
+            return Optional.empty();
         }
 
-        StringBuilder resultStringBuilder = new StringBuilder(result);
+        StringBuilder resultStringBuilder = new StringBuilder(result.get());
         resultStringBuilder.append("/vod/");
         resultStringBuilder.append(videoId);
         resultStringBuilder.append("/index");
         if (isLocal()) {
             resultStringBuilder.append("_local");
         }
         resultStringBuilder.append(".m3u8");
-        return resultStringBuilder.toString();
+        return Optional.of(resultStringBuilder.toString());
     }
 
-    @SuppressWarnings("null")
     private boolean isLocal() {
-        return (module == null || module.getIsLocal() == null) ? false : module.getIsLocal();
+        return getModule().map(NAWelcomeCamera::getIsLocal).orElse(false);
+    }
+
+    private void switchVideoSurveillance(boolean isOn) {
+        Optional<String> localCameraURL = getLocalCameraURL();
+        if (localCameraURL.isPresent()) {
+            String url = localCameraURL.get() + STATUS_CHANGE_URL_PATH + "?status=";
+            if(isOn) {
+                url += "on";
+            } else {
+                url += "off";
+            }
+            executeGETRequest(url);
+
+            invalidateParentCacheAndRefresh();
+        }
+    }
+
+    protected Optional<String> getLocalCameraURL() {
+        Optional<String> vpnURLOptional = getVpnUrl();
+        if (vpnURLOptional.isPresent()) {
+            final String vpnURL = vpnURLOptional.get();
+
+            //The local address is (re-)requested when it wasn't already determined or when the vpn address was changed.
+            if (!cameraAddress.isPresent() || cameraAddress.get().isVpnURLChanged(vpnURL)) {
+                Optional<JSONObject> json = executeGETRequestJSON(vpnURL + PING_URL_PATH);
+                cameraAddress = json.map(j -> j.optString("local_url", null))
+                        .map(localURL -> new CameraAddress(vpnURL, localURL));
+            }
+        }
+        return cameraAddress.map(CameraAddress::getLocalURL);
+    }
+
+    private Optional<JSONObject> executeGETRequestJSON(String url) {
+        try {
+            return executeGETRequest(url).map(JSONObject::new);
+        } catch (JSONException e) {
+            logger.warn("Error on parsing the content as JSON!", e);
```

Comment:
A DEBUG log will be sufficient I think.

Reply:
Debug? When this error occurs, it is an error, the functionality is broken when it occurs. Actually warning isn't enough.

Reply:
My general rule for logging levels is:
* trace - used for normal expected execution paths
* debug - used for unexepected execution paths, but well within normal operation
* info - used for notable points in an execution path, like a milestone. (in openhab we try reserve this logging level for the core, so bindings should rarely ever call this.)
* warn - used for notable unexpected execution paths that a regular user (not just a developer) should be notified of. Warnings should be used to indicate that something not-normal occurred and user intervention is required to resolve. Warnings do not indicate a failure to operate merely an abnormal condition of operation that can still be handled by the binding. Failures in binding operation should be indicated by changing the thing status to offline.
* error - used to indicate catastrophic program failure. This should be used to indicate a catastrophic failure in openhab's ability to operate. A failure in a binding would never cause openhab as a whole to fail so a failure in a bindings should never log an error. Instead that failure should be indicated by changing the thing status.

Now if a JSONException does occur here it doesn't really reflect a failure in the binding so much a failure in an external system and can be thought of more or less akin to connection loss. If you think that such a json parsing failure should be elevated above "warn" then I suggest changing the thing status to OFFLINE to reflect that.